src/Transition.js

```diff
@@ -326,7 +326,7 @@ class Transition extends React.Component {
     this.setNextCallback(handler)
 
     const doesNotHaveTimeoutOrListener = timeout == null && !this.props.addEndListener
-    if (!node || doesNotHaveTimeoutOrListener) {
+    if (doesNotHaveTimeoutOrListener) {
```

Comment:
Has its own commit (87832e91489d046012006c16cad58bd216cb40b2). No test broke upon removing this so I need some guidance to understand this condition.

Reply:
https://github.com/reactjs/react-transition-group/pull/464

Reply:
That PR did not introduce the evaluation of `node`.